clang/lib/Format/UnwrappedLineFormatter.cpp

```diff
@@ -1355,6 +1362,10 @@ unsigned UnwrappedLineFormatter::format(
     bool FixIndentation = (FixBadIndentation || ContinueFormatting) &&
                           Indent != TheLine.First->OriginalColumn;
     bool ShouldFormat = TheLine.Affected || FixIndentation;
+
+    if (Style.IgnorePPDefinitions && lineContainsPPDefinition(TheLine))
+      ShouldFormat = false;
```

Comment:
```suggestion
    bool ShouldFormat = (TheLine.Affected || FixIndentation) && (!Style.IgnorePPDefinitions || lineContainsPPDefinition(TheLine));
```
Maybe? Don't know which is better.

Reply:
Personally, I like the original approach but maybe having longer conditions is more in line with the rest of the code. Let me know if I should change it.

Reply:
I will not request the change. :)
I'm fine with either way.